packages/block-library/src/footnotes/format.js

```diff
@@ -86,19 +100,8 @@ export const format = {
 					onChange( newValue );
 				}
 
-				// BFS search to find the first footnote block.
-				let fnBlock = null;
-				{
-					const queue = [ ...getBlocks() ];
-					while ( queue.length ) {
-						const block = queue.shift();
-						if ( block.name === name ) {
-							fnBlock = block;
-							break;
-						}
-						queue.push( ...block.innerBlocks );
-					}
-				}
+				// Finds the first footnote block.
```

Comment:
This is interesting.

Pursuant to the [comment above](https://github.com/WordPress/gutenberg/pull/53003/files#r1275543962), `getEditedEntityRecord` _kinda_ works:

```js
				const blocks = getEditedEntityRecord(
					'postType',
					postType,
					postId
				)?.blocks;

				// Finds the first footnote block.
				let fnBlock = blocks?.find( ( block ) => block.name === name );
```

But first we have to interact with the page itself, e.g., by editing the content in any way



https://github.com/WordPress/gutenberg/assets/6458278/56b4f56e-aeab-4784-9832-4cbb1741c945




So we're getting closer 😄 

Is there a way to trigger this interactivity programmatically? 🤷🏻

Reply:
I can't say I'm hugely knowledgeable about the core-data stuff either, but would `getEntityRecord` work for retrieving the initial state?

Reply:
I think part of the problem might be that in order for the `selectionChange()` call below to work, we might need to somehow get the same list of blocks (and therefore clientIds) that gets set within the parent post content block's `useInnerBlocks` [here](https://github.com/WordPress/gutenberg/blob/fbc67a11105e5884c05a8498b01c70173ad181ee/packages/block-library/src/post-content/edit.js#L70) (the post content block's controlled inner blocks).

From playing around locally, when I've attempted to call `getEditedEntityRecord` or `getEntityRecord` to grab the data I need, we wind up with a duplicate of the blocks, rather than a reference to the existing Footnotes block in the editor 🤔